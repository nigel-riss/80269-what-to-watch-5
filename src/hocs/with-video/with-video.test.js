import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import withVideo from './with-video.jsx';


const MockComponent = (props) => {
  const {
    children,
    renderVideo,
  } = props;

  return (
    <React.Fragment>
      {children}
      {renderVideo({
        src: ``,
        poster: ``,
        height: 100,
        width: 200,
        isPlaying: false,
      })}
    </React.Fragment>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  renderVideo: PropTypes.func.isRequired,
};

const MockComponentWrapped = withVideo(MockComponent);


it(`Should  is rendered correctly`, () => {
  const tree = renderer
    .create((
      <MockComponentWrapped/>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
