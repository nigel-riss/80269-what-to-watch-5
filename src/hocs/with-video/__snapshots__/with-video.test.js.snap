// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should  is rendered correctly 1`] = `
<video
  disabled={true}
  height={100}
  muted={true}
  poster=""
  width={200}
/>
`;
